src/ring_theory/polynomial/symmetric.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2020 Hanting Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hanting Zhang, Johan Commelin
+-/
+import tactic
+import data.mv_polynomial.rename
+import data.mv_polynomial.comm_ring
+
+/-!
+# Symmetric Polynomials and Elementary Symmetric Polynomials
+
+This file defines symmetric polynomials and elementary symmetric polynomials.
+We also prove some basic facts about them.
+
+## Main declarations
+
+* `mv_polynomial.is_symmetric`
+
+* `mv_polynomial.esymm`
+
+## Notation
+
++ `esymm σ R n`, is the `n`th elementary symmetric polynomial in `mv_polynomial σ R`.
+
+As in other polynomial files, we typically use the notation:
+
++ `σ τ α : Type*` (indexing the variables)
+
++ `R S : Type*` `[comm_semiring R]` `[comm_semiring S]` (the coefficients)
+
++ `s : σ →₀ ℕ`, a function from `σ` to `ℕ` which is zero away from a finite set.
+This will give rise to a monomial in `mv_polynomial σ R` which mathematicians might call `X^s`
+
++ `r : R` elements of the coefficient ring
+
++ `i : σ`, with corresponding monomial `X i`, often denoted `X_i` by mathematicians
+
++ `p : mv_polynomial σ α`
+
+-/
+
+open equiv (perm)
+open_locale big_operators
+noncomputable theory
+
+namespace mv_polynomial
+
+variables {σ : Type*} {R : Type*}
+variables {τ : Type*} {S : Type*}
+
+/-- A mv_polynomial φ is symmetric if it is invariant under
+permutations of its variables by the  `rename` operation -/
+def is_symmetric [comm_semiring R] (φ : mv_polynomial σ R) : Prop :=
+∀ e : perm σ, rename e φ = φ
+
+namespace is_symmetric
+variables [comm_semiring R] [comm_semiring S] {φ ψ : mv_polynomial σ R}
+
+@[simp]
+lemma C (r : R) : is_symmetric (C r : mv_polynomial σ R) :=
+λ e, rename_C e r
+
+@[simp]
+lemma zero : is_symmetric (0 : mv_polynomial σ R) :=
+by { rw [← C_0], exact is_symmetric.C 0 }
+
+@[simp]
+lemma one : is_symmetric (1 : mv_polynomial σ R) :=
+by { rw [← C_1], exact is_symmetric.C 1 }
+
+@[simp]
+lemma add (hφ : is_symmetric φ) (hψ : is_symmetric ψ) : is_symmetric (φ + ψ) :=
+λ e, by rw [alg_hom.map_add, hφ, hψ]
+
+@[simp]
+lemma mul (hφ : is_symmetric φ) (hψ : is_symmetric ψ) : is_symmetric (φ * ψ) :=
+λ e, by rw [alg_hom.map_mul, hφ, hψ]
+
+@[simp]
+lemma smul (r : R) (hφ : is_symmetric φ) : is_symmetric (r • φ) :=
+λ e, by rw [alg_hom.map_smul, hφ]
+
+@[simp]
+lemma map (hφ : is_symmetric φ) (f : R →+* S) : is_symmetric (map f φ) :=
+λ e, by rw [← map_rename, hφ]
```

Comment:
I'm not sure these are good simp lemmas, as they're conditional on the hypothesis, and sometimes you want to prove `is_symmetric (φ + ψ)` but it is not true that `is_symmetric φ` and `is_symmetric ψ`.

Reply:
I agree, `add`, `mul`, `smul`, `neg`, `sub` shouldn't be simp-lemmas.
One can add a simp-lemma `neg_iff` that removes the `-`.